### PR TITLE
Login & Logout

### PR DESCRIPTION
**Pull request by awave1**

### CODE REVIEW

File: app/src/main/java/com/group15/fitnesstracker/db/FitnessTrackerDatabase.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    companion object {
+        @Volatile private var db: FitnessTrackerDatabase? = null
+
+        fun instance(context: Context): FitnessTrackerDatabase =
+                db ?: synchronized(this) {
+                    db ?: buildDb(context).also { db = it }
+                }
+
+        private fun buildDb(context: Context) =
+                Room.databaseBuilder(context, FitnessTrackerDatabase::class.java, DB_NAME).build()
```
Comment (dartuso) [inline app/src/main/java/com/group15/fitnesstracker/db/FitnessTrackerDatabase.kt]: Add .fallBackToDestructiveMigration
```suggestion
                Room.databaseBuilder(context, FitnessTrackerDatabase::class.java, DB_NAME).fallbackToDestructiveMigration().build()
```

File: app/src/main/java/com/group15/fitnesstracker/onboarding/login/LoginPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import android.content.Context
+import com.group15.fitnesstracker.R
+import com.group15.fitnesstracker.base.BaseView
+import com.group15.fitnesstracker.db.dao.UserDao
+import com.group15.fitnesstracker.util.Constants
+import io.reactivex.schedulers.Schedulers
+
+class LoginPresenter(val view: LoginContract.View, private val context: Context, private val userDao: UserDao): LoginContract.Presenter {
+    override fun start() {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
```
Comment (dartuso) [inline app/src/main/java/com/group15/fitnesstracker/onboarding/login/LoginPresenter.kt:13]: Necessary?

File: app/src/main/java/com/group15/fitnesstracker/onboarding/login/LoginPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @SuppressLint("CheckResult")
+    override fun login(username: String, password: String) {
+        // @TODO: implement password handling as well
```
Comment (dartuso) [inline app/src/main/java/com/group15/fitnesstracker/onboarding/login/LoginPresenter.kt:18]: Please implement zaddy

**Review by dartuso — CHANGES_REQUESTED** [review]
GJ pls review

File: app/src/main/java/com/group15/fitnesstracker/db/FitnessTrackerDatabase.kt (hunk 1 above)
Comment (awave1) [reply app/src/main/java/com/group15/fitnesstracker/db/FitnessTrackerDatabase.kt]: wow look at you mr.app developer

File: app/src/main/java/com/group15/fitnesstracker/onboarding/login/LoginPresenter.kt (hunk 2 above)
Comment (awave1) [reply app/src/main/java/com/group15/fitnesstracker/onboarding/login/LoginPresenter.kt:13]: yes, for now. `BasePresenter` has that method. Not really sure why i added it, but can be removed

**Pull request merged**